From d0eed5363d7ff1854ba7a13d8f146a08b2db6389 Mon Sep 17 00:00:00 2001
From: S A Muqthyar Ahmed <syed.abdul.muqthyar.ahmed@intel.com>
Date: Tue, 16 Jun 2026 02:46:51 +0530
Subject: [PATCH] backport: Disable VFIO_PCI_DMABUF for kernel 5.15

Errors:
- vfio_pci_dmabuf.c:36:47: error: invalid use of undefined type 'struct dma_buf'
- vfio_pci_dmabuf.c:58:16: error: implicit declaration of function 'dma_buf_phys_vec_to_sgt'
- vfio_pci_dmabuf.c:88:21: error: variable 'vfio_pci_dmabuf_ops' has initializer but incomplete type

VFIO_PCI_DMABUF requires APIs and structures not available in kernel 5.15:
- struct dma_buf and dma_buf_ops have different members
- dma_buf_phys_vec_to_sgt() function doesn't exist
- struct phys_vec is not defined

Disable this feature for kernel 5.15 by setting config to 'def_bool n'.

Signed-off-by: S A Muqthyar Ahmed <syed.abdul.muqthyar.ahmed@intel.com>
---
 drivers/vfio/pci/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/vfio/pci/Kconfig b/drivers/vfio/pci/Kconfig
index 4985296..cd9faa7 100644
--- a/drivers/vfio/pci/Kconfig
+++ b/drivers/vfio/pci/Kconfig
@@ -58,7 +58,7 @@ config VFIO_PCI_ZDEV_KVM
 	  To enable s390x KVM vfio-pci extensions, say Y.
 
 config VFIO_PCI_DMABUF
-	def_bool y if VFIO_PCI_CORE && PCI_P2PDMA && DMA_SHARED_BUFFER
+	def_bool n
 
 source "drivers/vfio/pci/mlx5/Kconfig"
 
-- 
2.43.0

